libs/community/langchain_community/chat_models/oci_generative_ai.py

```diff
@@ -0,0 +1,363 @@
+import json
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterator, List, Mapping, Optional, Sequence
+
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models.chat_models import (
+    BaseChatModel,
+    generate_from_stream,
+)
+from langchain_core.messages import (
+    AIMessage,
+    AIMessageChunk,
+    BaseMessage,
+    ChatMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain_core.outputs import ChatGeneration, ChatGenerationChunk, ChatResult
+from langchain_core.pydantic_v1 import Extra
+
+from langchain_community.llms.oci_generative_ai import OCIGenAIBase
+from langchain_community.llms.utils import enforce_stop_tokens
```

Comment:
does `oci` support this through the sdk, or is this just a precaution?

Reply:
for some models it is supported through the OCI SDK for some not